test/lib/tls_certificates/test_ca_client.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import json
+import yaml
+import cryptography
+
+from ops import framework
+from ops.charm import CharmBase
+from ops import testing
+
+from ops.lib.tls_certificates import ca_client
+from pathlib import Path
+
+
+class TestCAClient(unittest.TestCase):
+
+    def setUp(self):
+        self.harness = testing.Harness(CharmBase, meta='''
+            name: myserver
+            peers:
+              ca-client:
+                interface: tls-certificates
+        ''')
+
+        self.harness.begin()
+        self.ca_client = ca_client.CAClient(self.harness.charm, 'ca-client')
+
+    def test_is_join(self):
```

Comment:
is_joined

though I think this is as much a test of ca_available as it is of is_joined.

Reply:
`test_ca_available` might be better then

Reply:
https://github.com/canonical/operator/pull/210/commits/7407ab5ebd35c1cf52547f399c447e0a6f758bac